main.tf

```diff
@@ -71,7 +71,7 @@ resource "aws_rds_cluster" "this" {
   copy_tags_to_snapshot               = var.copy_tags_to_snapshot
   database_name                       = var.is_primary_cluster ? var.database_name : null
   db_cluster_instance_class           = var.db_cluster_instance_class
-  db_cluster_parameter_group_name     = var.create_db_cluster_parameter_group ? aws_rds_cluster_parameter_group.this[0].id : local.cluster_parameter_group_name
```

Comment:
this was the offending line which led to the issue - I incorrectly used the local value which causes an issue where it won't default to an empty/null value when users do not provide a name which signals to the service to use the default parameter group for the given engine/family version